test/test_parser.rb

```diff
@@ -4976,10 +4976,11 @@ def test_bom
     end
 
     def test_magic_encoding_comment
+      funny_russian_sym = "проверка".encode(Encoding::KOI8_R).to_sym
       assert_parses(
         s(:begin,
-          s(:lvasgn, :"проверка", s(:int, 42)),
-          s(:send, nil, :puts, s(:lvar, :"проверка"))),
+          s(:lvasgn, funny_russian_sym, s(:int, 42)),
+          s(:send, nil, :puts, s(:lvar, funny_russian_sym))),
```

Comment:
The test file it itself encoded in UTF-8. The only way I know to get a symbol in a different encoding is to convert a string over manually and then `to_sym` it.